### PR TITLE
Don't clip PowerNorm inputs < vmin

### PR DESCRIPTION
**Pull request by dstansby**
## PR summary
This updates the behaviour of `PowerNorm` to linearly normalize input values < vmin, instead of clipping them.

Fixes https://github.com/matplotlib/matplotlib/issues/25239; replaces https://github.com/matplotlib/matplotlib/pull/25256. See discussion at https://github.com/matplotlib/matplotlib/pull/25256 for more context on this.

## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [n/a] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [x] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Review by tacaswell — APPROVED** [review]

File: lib/matplotlib/tests/test_colors.py [hunk 1]
```diff
@@ -555,7 +555,7 @@ def test_PowerNorm():
     assert_array_almost_equal(norm(a), pnorm(a))
 
     a = np.array([-0.5, 0, 2, 4, 8], dtype=float)
-    expected = [0, 0, 1/16, 1/4, 1]
+    expected = [-1/16, 0, 1/16, 1/4, 1]
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_colors.py:558]: What happens to the inverse? On line 563, I see that it is tested, but excludes the value below `vmin`. But that was when it became 0 for everything, which could never be inverted uniquely; is it uniquely invertible now?

Comment (dstansby) [reply lib/matplotlib/tests/test_colors.py:558]: Ah you're very correct that I didn't change the inverse, I've marked this as draft again so it doesn't get merged until I fix that.

File: lib/matplotlib/tests/test_colorbar.py [hunk 2]
```diff
@@ -552,10 +552,10 @@ def test_colorbar_lognorm_extension(extend):
 
 def test_colorbar_powernorm_extension():
     # Test that colorbar with powernorm is extended correctly
-    f, ax = plt.subplots()
-    cb = Colorbar(ax, norm=PowerNorm(gamma=0.5, vmin=0.0, vmax=1.0),
-                  orientation='vertical', extend='both')
-    assert cb._values[0] >= 0.0
```
Comment (dstansby) [inline lib/matplotlib/tests/test_colorbar.py:558]: This check came in with https://github.com/matplotlib/matplotlib/pull/11610, but from that PR it seems like it's enough just to add the colorbar and check no error or warning is raised. It's expected that `_values` now has a value < 0 with this PR, so I don't think it's meaningful to do this check any more.

File: lib/matplotlib/tests/test_colors.py (hunk 1 above)
Comment (dstansby) [reply lib/matplotlib/tests/test_colors.py:558]: Thanks for catching, should be fixed now with an updated test to make sure the inverse is tested.

File: lib/matplotlib/colors.py [hunk 3]
```diff
@@ -1969,6 +1969,8 @@ class PowerNorm(Normalize):
     .. math::
 
         \left ( \frac{x - v_{min}}{v_{max}  - v_{min}} \right )^{\gamma}
+
+    For input values below *vmin*, gamma is set to zero.
```
Comment (tacaswell) [inline lib/matplotlib/colors.py:1973]:
```suggestion
    For input values below *vmin*, gamma is set to one.
```

**Comment by tacaswell** [comment]
I think going with linear is better because it is invertable, but could we have also fixed the colorbar issue more narrowly by setting the under values to `-1` instead of 0?

File: lib/matplotlib/tests/test_colors.py [hunk 4]
```diff
@@ -591,6 +595,15 @@ def test_PowerNorm_translation_invariance():
     assert_array_almost_equal(pnorm(a - 2), expected)
 
 
+def test_powernorm_cbar_limits():
+    fig, ax = plt.subplots()
+    vmin, vmax = 300, 1000
+    data = np.arange(100*100).reshape(100, 100) + vmin
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_colors.py:601]: Do we need 10000 points? That seems a bit large, though I didn't check how slow the test was.

**Comment by dstansby** [comment]
> I think going with linear is better because it is invertable, but could we have also fixed the colorbar issue more narrowly by setting the under values to `-1` instead of 0?

Yes, setting everything under to -1 also fixes the colorbar issue.

**Review by tacaswell — APPROVED** [review]

**Review by QuLogic — APPROVED** [review]

**Pull request merged**